### PR TITLE
Report Malicious IPs

### PR DESCRIPTION
**Pull request by krrish-sehgal**
**Fixes**: #2591

## Description

This PR introduces a feature that allows users to report suspicious IP addresses with a description and title, targeting IPs involved in unauthorized activities such as phishing, scanning, and attacks.

### Changes Implemented

- **Model**: 
  - `ReportIP` model with migration to store user-reported IP addresses and other details.
- **Forms**: 
  - `IpReportForm` to facilitate user submissions of IP reports.
- **Views**: 
  - **`ReportIpView`**: Handles GET and POST requests to display and process the report IP form.
  - **`MaliciousIpListView`**: Displays a list of all IP addresses reported by users for easy tracking and review.
- **Navigation**:
  - Added options in the side and top navbars to access this feature.
- **python version made generic**

### New Templates Created

- `report_ip.html`: Renders the form for users to report IPs.
- `malicious_ips_list.html`: Lists all IPs reported for malicious activities.

### More features that can be added in further updates: 
- `Like and Dislike Feature: Implementing a feedback mechanism to allow users to express their opinions on reported IPs.`
-`Duplicate Check: Adding this button in from to allow users to browse through the IPs.`
-`IP Search Functionality: Developing a search feature to allow users to easily find specific reported IPs.`

### Attached Demo
[Video demo of the feature](https://github.com/user-attachments/assets/db12fb47-da9b-4e78-8ae7-1bef7b05fe76)

### CODE REVIEW

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Looks good thank you can you please make a few tweaks?

**Comment by krrish-sehgal** [comment]
I have made the model and names in the form more generic and have addressed all the changes that were requested including allowing users to select the type of activity ie (friendly or malicious).

I have also included a ip validation from the backend as well now.

The new reported_ips_list page looks something like this : 
Screenshot 2024-11-07 at 3 46 52 PM

**Review by DonnieBLT — APPROVED** [review]

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Can you please check the code scanning alerts?

**Comment by krrish-sehgal** [comment]
> Can you please check the code scanning alerts?

Sure, on it.

**Review by DonnieBLT — DISMISSED** [review]

**Comment by krrish-sehgal** [comment]
the conflicts have been resolved in my last commit , thanks.

**Review by DonnieBLT — DISMISSED** [review]

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
We are no longer using the contributor stats table

**Review by DonnieBLT — DISMISSED** [review]

**Comment by krrish-sehgal** [comment]
@DonnieBLT , any feedback as to why this test case always fails?

**Comment by krrish-sehgal** [comment]
Could you kindly merge the current changes before proceeding with any further merges? This will help in avoiding any potential conflicts.

**Review by DonnieBLT — CHANGES_REQUESTED** [review]

File: website/templates/includes/header.html
[the remark this answers is not in the document]
Comment (krrish-sehgal) [reply website/templates/includes/header.html:1]: sure sir , on it.

**Review by DonnieBLT — CHANGES_REQUESTED** [review]

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
One tweak

**Comment by krrish-sehgal** [comment]
> One tweak

Sir , you mentioned that the report-ip button would be good on a page , not the side-nav , could you specify which page you are referring ?  Is it the home page like i mentioned before ?

**Comment by krrish-sehgal** [comment]
sure , got it.

**Comment by krrish-sehgal** [comment]
@
Screenshot 2024-11-13 at 5 53 12 PM

this button will appear on this page only and looks clean.

**Review by DonnieBLT — APPROVED** [review]

**Pull request merged**